src/tinyexpr.cpp

```diff
@@ -199,53 +195,63 @@ static double min(double a, double b) {
-    {L"cosh", reinterpret_cast<void *>(static_cast<te_fun1>(std::cosh)), TE_FUNCTION1},
-    {L"e", reinterpret_cast<void *>(static_cast<te_fun0>(e)), TE_FUNCTION0},
-    {L"exp", reinterpret_cast<void *>(static_cast<te_fun1>(std::exp)), TE_FUNCTION1},
-    {L"fac", reinterpret_cast<void *>(static_cast<te_fun1>(fac)), TE_FUNCTION1},
-    {L"floor", reinterpret_cast<void *>(static_cast<te_fun1>(std::floor)), TE_FUNCTION1},
-    {L"ln", reinterpret_cast<void *>(static_cast<te_fun1>(std::log)), TE_FUNCTION1},
-    {L"log", reinterpret_cast<void *>(static_cast<te_fun1>(std::log10)), TE_FUNCTION1},
-    {L"log10", reinterpret_cast<void *>(static_cast<te_fun1>(std::log10)), TE_FUNCTION1},
-    {L"log2", reinterpret_cast<void *>(static_cast<te_fun1>(std::log2)), TE_FUNCTION1},
-    {L"max", reinterpret_cast<void *>(static_cast<te_fun2>(max)), TE_FUNCTION2},
-    {L"min", reinterpret_cast<void *>(static_cast<te_fun2>(min)), TE_FUNCTION2},
-    {L"ncr", reinterpret_cast<void *>(static_cast<te_fun2>(ncr)), TE_FUNCTION2},
-    {L"npr", reinterpret_cast<void *>(static_cast<te_fun2>(npr)), TE_FUNCTION2},
-    {L"pi", reinterpret_cast<void *>(static_cast<te_fun0>(pi)), TE_FUNCTION0},
-    {L"pow", reinterpret_cast<void *>(static_cast<te_fun2>(std::pow)), TE_FUNCTION2},
-    {L"round", reinterpret_cast<void *>(static_cast<te_fun1>(std::round)), TE_FUNCTION1},
-    {L"sin", reinterpret_cast<void *>(static_cast<te_fun1>(std::sin)), TE_FUNCTION1},
-    {L"sinh", reinterpret_cast<void *>(static_cast<te_fun1>(std::sinh)), TE_FUNCTION1},
-    {L"sqrt", reinterpret_cast<void *>(static_cast<te_fun1>(std::sqrt)), TE_FUNCTION1},
-    {L"tan", reinterpret_cast<void *>(static_cast<te_fun1>(std::tan)), TE_FUNCTION1},
-    {L"tanh", reinterpret_cast<void *>(static_cast<te_fun1>(std::tanh)), TE_FUNCTION1},
-    {L"tau", reinterpret_cast<void *>(static_cast<te_fun0>(tau)), TE_FUNCTION0},
+static constexpr double kMaximumContiguousInteger =
+    double(1LLU << std::numeric_limits<double>::digits) - 1;
+
+static double maximum(const std::vector<double> &args) {
+    double ret = -kMaximumContiguousInteger;
+    for (auto a : args) ret = max(ret, a);
+    return ret;
+}
+
+static double minimum(const std::vector<double> &args) {
+    double ret = kMaximumContiguousInteger;
+    for (auto a : args) ret = min(ret, a);
+    return ret;
+}
+
+static const std::unordered_map<std::wstring, te_fun_t> functions = {
+    // clang-format off
+    {L"abs", std::fabs},
+    {L"acos", std::acos},
+    {L"asin", std::asin},
+    {L"atan", std::atan},
+    {L"atan2", std::atan2},
+    {L"bitand", bit_and},
+    {L"bitor", bit_or},
+    {L"bitxor", bit_xor},
+    {L"ceil", std::ceil},
+    {L"cos", std::cos},
+    {L"cosh", std::cosh},
+    {L"e", M_E},
+    {L"exp", std::exp},
+    {L"fac", fac},
+    {L"floor", std::floor},
+    {L"ln", std::log},
+    {L"log", std::log10},
+    {L"log10", std::log10},
+    {L"log2", std::log2},
+    {L"max", max},
+    {L"maximum", maximum},
```

Comment:
These should be one function `max` that is variadic. Same for `min`.

At least I can't see any point for having both.